src/solvers/dgsem_tree/indicators_2d.jl

```diff
@@ -91,40 +92,100 @@ function (indicator_hg::IndicatorHennemannGassner)(u::AbstractArray{<:Any,4},
-      alpha[left]  = max(alpha_tmp[left],  0.5 * alpha_tmp[right], alpha[left])
-      alpha[right] = max(alpha_tmp[right], 0.5 * alpha_tmp[left],  alpha[right])
-    end
 
-    # Loop over L2 mortars
-    for mortar in eachmortar(dg, cache)
-      # Get neighboring element ids
-      lower = cache.mortars.neighbor_ids[1, mortar]
-      upper = cache.mortars.neighbor_ids[2, mortar]
-      large = cache.mortars.neighbor_ids[3, mortar]
-
-      # Apply smoothing
-      alpha[lower] = max(alpha_tmp[lower], 0.5 * alpha_tmp[large], alpha[lower])
-      alpha[upper] = max(alpha_tmp[upper], 0.5 * alpha_tmp[large], alpha[upper])
-      alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[lower], alpha[large])
-      alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[upper], alpha[large])
-    end
+    apply_smoothing!(u, mesh, equations, dg, alpha, alpha_tmp, cache)
+
   end
 
   return alpha
 end
 
 
+function apply_smoothing!(u::AbstractArray{<:Any, 4}, mesh::Union{TreeMesh{2}, P4estMesh{2}}, equations, dg::DGSEM, alpha, alpha_tmp, cache)
+
+  # Diffuse alpha values by setting each alpha to at least 50% of neighboring elements' alpha
+  # Copy alpha values such that smoothing is indpedenent of the element access order
+  alpha_tmp .= alpha
+
+  # Loop over interfaces
+  for interface in eachinterface(dg, cache)
+    # Get neighboring element ids
+    left  = cache.interfaces.neighbor_ids[1, interface]
+    right = cache.interfaces.neighbor_ids[2, interface]
+
+    # Apply smoothing
+    alpha[left]  = max(alpha_tmp[left],  0.5 * alpha_tmp[right], alpha[left])
+    alpha[right] = max(alpha_tmp[right], 0.5 * alpha_tmp[left],  alpha[right])
+  end
+
+  # Loop over L2 mortars
+  for mortar in eachmortar(dg, cache)
+    # Get neighboring element ids
+    lower = cache.mortars.neighbor_ids[1, mortar]
+    upper = cache.mortars.neighbor_ids[2, mortar]
+    large = cache.mortars.neighbor_ids[3, mortar]
+
+    # Apply smoothing
+    alpha[lower] = max(alpha_tmp[lower], 0.5 * alpha_tmp[large], alpha[lower])
+    alpha[upper] = max(alpha_tmp[upper], 0.5 * alpha_tmp[large], alpha[upper])
+    alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[lower], alpha[large])
+    alpha[large] = max(alpha_tmp[large], 0.5 * alpha_tmp[upper], alpha[large])
+  end
+  
+end
+
+
+function apply_smoothing!(u::AbstractArray{<:Any, 4}, mesh::StructuredMesh{2}, equations, dg::DGSEM, alpha, alpha_tmp, cache)
+
+  # Diffuse alpha values by setting each alpha to at least 50% of neighboring elements' alpha
+  # Copy alpha values such that smoothing is indpedenent of the element access order
+  alpha_tmp .= alpha
+
+  # Loop over elements, because there is no interface container
+  for element in eachelement(dg,cache)
+    # Get neighboring element ids
+    left  = cache.elements.left_neighbors[1, element]
+    right = cache.elements.right_neighbors[1, element]
+    up    = cache.elements.left_neighbors[2, element] 
+    down  = cache.elements.right_neighbors[2, element]
```

Comment:
```suggestion
    down = cache.elements.left_neighbors[2, element] 
    up   = cache.elements.right_neighbors[2, element]
```
If this uses the same directional notation as the `TreeMesh`, these should be flipped.